sdks/java/core/src/main/java/org/apache/beam/sdk/runners/PTransformOverrideFactory.java

```diff
@@ -49,6 +51,15 @@
   InputT getInput(List<TaggedPValue> inputs, Pipeline p);
 
   /**
+   * Returns a {@link Map} from the expanded values in {@code newOutput} to the values produced by
+   * the original transform.
+   *
+   * <p>Every element in {@code outputs} must be present exactly once within the returned map.
```

Comment:
(you said this to me out loud just now): "Not necessarily, because if you know they are the same semantic thing, you can merge them"